### PR TITLE
Add sapuxlayer to package json 

### PR DESCRIPTION
**Pull request by kjose90**
https://github.com/SAP/open-ux-tools/issues/2326

- added `sapuxType` to `ui5-application-writer` package interface
- added tests

### CODE REVIEW

File: packages/ui5-application-writer/src/types.ts [hunk 1]
```diff
@@ -9,6 +10,7 @@ export interface Package {
     ui5?: {
         dependencies?: string[];
     };
+    sapuxLayer?: UI5FlexLayer;
```
Comment (IainSAP) [inline packages/ui5-application-writer/src/types.ts:13]: 👍

**Review by IainSAP — APPROVED** [review]
Thx @kjose90 

- Cleanly adds support for sapuxlayer property
- Adds tests to cover

File: packages/ui5-application-writer/src/types.ts (hunk 1 above)
Comment (IainSAP) [reply packages/ui5-application-writer/src/types.ts:13]: Would you please add a comment to the property explaining its use and default assignment.

**Review by longieirl — APPROVED** [review]
Changes look good.
Appropiate tests to cover change.
No chanageset required, being merged into a larger PR.

File: packages/ui5-application-writer/src/types.ts [hunk 2]
```diff
@@ -9,6 +10,13 @@ export interface Package {
     ui5?: {
         dependencies?: string[];
     };
+    /**
+     * Defines the SAPUI5 flexibility layer to manage smart control changes and provide personalization functions for your SAPUI5 applications.
+     *
+     * - `VENDOR`: Stores SAP-delivered content.
+     * - `CUSTOMER_BASE`: Stores customer-specific changes.
+     */
+    sapuxLayer?: UI5FlexLayer;
```
Comment (Klaus-Keller) [inline packages/ui5-application-writer/src/types.ts:19]: There is already a `devDependecy` to `@sap-ux/project-access` in this package, so please use existing type `Package`:

https://github.com/SAP/open-ux-tools/blob/875972fc945d069a149ed0e7d06eb0f763790900/packages/project-access/src/types/package/index.ts#L12

**Review by Klaus-Keller — CHANGES_REQUESTED** [review]

File: packages/ui5-application-writer/src/types.ts (hunk 2 above)
Comment (kjose90) [reply packages/ui5-application-writer/src/types.ts]: Thanks @Klaus-Keller  - using Package type from  `@sap-ux/project-access` as suggested.

Comment (Klaus-Keller) [reply packages/ui5-application-writer/src/types.ts]: Thanks

**Review by Klaus-Keller — APPROVED** [review]

**Comment by Klaus-Keller** [comment]
> > Also changeset is missing
> 
> Its being merged into a large PR and PR is against a feature branch.

Missed that, my bad.

**Comment by kjose90** [comment]
Thanks for review

**Pull request merged**